toolkit.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+import sys
+import cv2 as cv
+import numpy as np
+import ntpath
+import functools
+import time
+
+
+def timeit(func):
+    """
+    Poors man profiling function to
+    measure time it takes to finish function.
+    """
+    @functools.wraps(func)
+    def newfunc(*args, **kwargs):
+        start_time = time.time()
+        out = func(*args, **kwargs)
+        elapsed_time = time.time() - start_time
+        msg = 'function [{}] finished in {} ms'
+        print(msg.format(func.__name__, int(elapsed_time * 1000)))
+        return out
+    return newfunc
+
+
+class BoardToolkit:
+    def __init__(self, lo_thresh=100, hi_thresh=255):
+        self.lo_thresh = lo_thresh
+        self.hi_thresh = hi_thresh
+
+    def __get_index(self, arr):
+        """
+        Function for returning min, max tuples from
+        our arrays.
+        """
+        max = np.argmax(arr)
+        min = np.argmin(arr)
+        return (min, max)
+
+    def calc_hist(self, image):
+        """
+        Function gets rid of low intensity pixels values in
+        color spectrum and sets them to 0. Then it calculates
+        histogram of green color in x and y axis from image.
+        Then we calculate 1-th order difference on this
+        histogram and return min and max value indexes.
+        """
+        (width, height, _) = image.shape
+        vlines = np.zeros(width)
+        hlines = np.zeros(height)
+        image[:, :, 1] = cv.inRange(image[:, :, 1],
+                                    self.lo_thresh,
+                                    self.hi_thresh)
+        vlines = cv.reduce(image[:, :, 1], 1,
+                           cv.cv.CV_REDUCE_SUM,
+                           dtype=cv.CV_32S)
+        hlines = cv.reduce(image[:, :, 1], 0,
+                           cv.cv.CV_REDUCE_SUM,
+                           dtype=cv.CV_32S)
+        y = np.diff(map(lambda *row: list(row), *vlines)[0])
```

Comment:
@Foxiy this is a nasty hack and while it's clear to me what it does it should be still commented.